python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -630,6 +640,18 @@ def get_asset_provenance(self, asset_key: AssetKey) -> Optional[DataProvenance]:
             None if record is None else extract_data_provenance_from_entry(record.event_log_entry)
         )
 
+    @public
+    def set_data_version(self, asset_key: AssetKey, data_version: DataVersion) -> None:
+        """Set the data version for an asset being materialized by the currently executing step.
+        This is mostly useful for external execution situations where it is not possible to return
+        an `Output`.
+
+        Args:
+            asset_key (AssetKey): Key of the asset for which to set the data version.
+            data_version (DataVersion): The data version to set.
+        """
+        self._step_execution_context.set_data_version(asset_key, data_version)
+
```

Comment:
Added to support reporting the data version back from remote execution (there is no `Output` object, where we would normally set the `data_version`).